api/v1beta1/gitrepository_types.go

```diff
@@ -101,6 +101,12 @@ type GitRepositoryRef struct {
 	// The Git commit SHA to checkout, if specified Tag filters will be ignored.
 	// +optional
 	Commit string `json:"commit,omitempty"`
+
+	// When enabled, after the clone is created, initializes all submodules within,
+	// using their default settings.
+	// This option is available only when using the 'go-git' GitImplementation.
+	// +optional
+	IncludeSubmodules bool `json:"includeSubmodules,omitempty"`
```

Comment:
Why here rather than at the top? It doesn't really pertain to the reference being checked out.

Reply:
The current Git package doesn't take the whole spec but only the ref, that's why I opted to include it here. I can change the interfaces but I don't see an issue of this being under ref, it reads as "clone this <branch,tag,commit> including submodules".

Reply:
I moved it to spec